WcaOnRails/app/views/media/_media_table.html.erb

```diff
@@ -0,0 +1,95 @@
+<% show_edit_buttons = defined?(show_edit_buttons) && show_edit_buttons %>
+
+<script>
+  function sortBySorter(a, b) {
+    return b['sort-by'] - a['sort-by'];
+  }
+</script>
+
+<form action="" method="get" class="form-inline">
+  <div class="form-group">
+    <label for="year-selector"><%= t '.year' %></label>
+    <%= select_tag "year",
+                   year_option_tags(selected_year: params[:year], future: false),
+                   id: "year-selector",
+                   class: "form-control"
+    %>
+  </div>
+
+  <div class="form-group">
+    <%= label_tag(t('competitions.index.region')) %>
+    <%= select_tag(:region, region_option_tags(selected_id: params[:region]), class: "form-control") %>
+  </div>
+
+  <% if show_edit_buttons %>
+    <div class="form-group">
+      <label for="status-selector"><%= t 'activerecord.attributes.competition_medium.status' %></label>
+      <%= select_tag "status",
+                     options_for_select(CompetitionMedium.statuses_i18n.invert, params[:status]),
+                     id: "status-selector",
+                     class: "form-control"
+      %>
+    </div>
+  <% end %>
+
+  <input type="submit" class="btn btn-default" value="<%= t 'common.filter' %>" />
+</form>
+
+<%= wca_table table_class: "media", data: { toggle: "table", sort_name: "submission-date", sort_order: "desc" } do %>
+  <thead>
+    <tr>
+      <th data-sortable="true" data-field="submission-date" data-sort-name="_submission-date_data" data-sorter="sortBySorter">Submission Date</th>
+      <th data-sortable="true" data-field="competition-date" data-sort-name="_competition-date_data" data-sorter="sortBySorter">Competition Date</th>
+      <th><%= t 'activerecord.attributes.competition_medium.competitionId' %></th>
+      <th><%= t '.location' %></th>
+      <th><%= t 'activerecord.attributes.competition_medium.type' %></th>
+      <th><%= t 'activerecord.attributes.competition_medium.uri' %></th>
+      <th></th>
+
+      <!-- Extra column for .table-greedy-last-column -->
+      <th></th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <% media.each do |medium| %>
+      <% competition = medium.competition %>
+      <tr data-medium-id="<%= medium.id %>">
+        <td data-sort-by="<%= medium.timestampSubmitted.to_i %>"><%= medium.timestampSubmitted.to_formatted_s(:long_utc) %></td>
+        <td data-sort-by="<%= competition.start_date.to_time.to_i %>">
+          <%= wca_date_range(competition.start_date, competition.end_date) %>
+        </td>
+        <td><%= link_to competition.name, competition_path(competition) %></td>
+        <td><strong><%= competition.country.name %></strong>, <%= competition.cityName %></td>
+        <td><%= medium.type %></td>
+        <td><%= link_to medium.text, medium.uri %></td>
+        <% if show_edit_buttons %>
```

Comment:
If this is false then we have more header columns than the body ones.

Reply:
Whoops, good catch. I'll add an if statement up in the `thead` as well.